include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3123,6 +3123,242 @@ namespace internal
 
 
 
+      template <int dim, int spacedim, bool level_dof_access>
+      static bool
+      p_refine_flag_set(const DoFCellAccessor<DoFHandler<dim, spacedim>,
+                                              level_dof_access> &accessor)
+      {
+        (void)accessor;
+        // ::DoFHandler does not support p-refinement
+        Assert(
+          false,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
+        return false;
+      }
+
+
+
+      template <int dim, int spacedim, bool level_dof_access>
+      static bool
+      p_refine_flag_set(
+        const DoFCellAccessor<dealii::hp::DoFHandler<dim, spacedim>,
+                              level_dof_access> &accessor)
+      {
+        Assert(
+          accessor.dof_handler != nullptr,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
```

Comment:
is this decltype thing what we do elsewhere in this file?

Reply:
There are a lot of type checks via `std::decay<decltype(accessor)>::type` in this file, so I thought I'd do it similarly.

I agree with replacing the exception in the assertion above, where we check if the `DoFHandler` is of type `hp::DoFHandler`. But for the other occasions, I would leave it as it is, at least for now.

Reply:
We check the assertion in a free function.

We could either use the `decltype` thing, or we have to specify the class.

Reply:
I just don't quite recall what we were doing with the `decltype` thing here. Let's not worry about it then.